Python/compile.c

```diff
@@ -4842,8 +4842,12 @@ maybe_optimize_method_call(struct compiler *c, expr_ty e)
     }
     /* Alright, we can optimize the code. */
     VISIT(c, expr, meth->v.Attribute.value);
-    int old_lineno = c->u->u_lineno;
-    c->u->u_lineno = meth->end_lineno;
+    SET_LOC(c, meth);
+    if (meth->lineno != meth->end_lineno) {
+        // Make start location match attribute
+        c->u->u_lineno = meth->end_lineno;
+        c->u->u_col_offset = meth->end_col_offset - (int)PyUnicode_GetLength(meth->v.Attribute.attr)-1;
```

Comment:
Once https://github.com/python/cpython/pull/93682 is merged with will need to change to something like:

```
struct location attr_loc = Location(meth->end_lineno,
                                    c->u->u_end_lineno,
                                    meth->end_col_offset - (int)PyUnicode_GetLength(meth->v.Attribute.attr)-1,
                                    c->u->u_end_col_offset);
c->u->u_loc = attr_loc;
```

Reply:
Sure, it’s a minor conflict anyway. Just the length of the attr can be calculated only once.